maker/src/main.rs

```diff
@@ -205,3 +225,22 @@ async fn main() -> Result<()> {
 
     Ok(())
 }
+
+/// Convenient type to load from toml
+#[derive(Deserialize)]
+struct BlockedPeers {
+    blocked: HashSet<PeerId>,
+}
+
+async fn load_blocked_peers(blocked_peers_path: &Path) -> Result<HashSet<PeerId>> {
+    if blocked_peers_path.try_exists()? {
+        let raw = tokio::fs::read_to_string(blocked_peers_path).await?;
+        Ok(toml::from_str::<BlockedPeers>(&raw)?.blocked)
+    } else {
+        tracing::info!(
+            ?blocked_peers_path,
+            "No blocked peers list found; ignoring and allowing all connections",
+        );
+        Ok(Default::default())
+    }
```

Comment:
there's no need for special handling here; when you call the function you also just allow all peers in case of any error

```rust
async fn load_blocked_peers(blocked_peers_path: &Path) -> Result<HashSet<PeerId>> {
    anyhow::ensure!(
        blocked_peers_path.try_exists(),
        "No blocked peers file found in {blocked_peers_path}",
    );
    let raw = tokio::fs::read_to_string(blocked_peers_path).await?;
    Ok(toml::from_str::<BlockedPeers>(&raw)?.blocked)
}
```